src/filters/prisma-exception.filter.ts

```diff
@@ -0,0 +1,107 @@
+  P2014 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** A related record could not be found. {details} */
+  P2015 = HttpStatus.NOT_FOUND,
+  /** Query interpretation error. {details} */
+  P2016 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** The records for relation {relation_name} between the {parent_name} and {child_name} models are not connected */
+  P2017 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** The required connected records were not found. {details} */
+  P2018 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Input error. {details} */
+  P2019 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Value out of range for the type. {details} */
+  P2020 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** The table {table} does not exist in the current database */
+  P2021 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** The column {column} does not exist in the current database */
+  P2022 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Inconsistent column data: {message} */
+  P2023 = HttpStatus.CONFLICT,
+  /** Timed out fetching a new connection from the connection pool. (More info: http://pris.ly/d/connection-pool (Current connection pool timeout: {timeout}, connection limit: {connection_limit}) */
+  P2024 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** An operation failed because it depends on one or more records that were required but not found. {cause} */
+  P2025 = HttpStatus.NOT_FOUND,
+  /** The current database provider doesn't support a feature that the query used: {feature} */
+  P2026 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Multiple errors occurred on the database during query execution: {errors} */
+  P2027 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Transaction API error: {error} */
+  P2028 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Cannot find a fulltext index to use for the search, try adding a @@fulltext([Fields...]) to your schema */
+  P2030 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Prisma needs to perform transactions, which requires your MongoDB server to be run as a replica set. See details: https://pris.ly/d/mongodb-replica-set */
+  P2031 = HttpStatus.CONFLICT,
+  /** A number used in the query does not fit into a 64 bit signed integer. Consider using BigInt as field type if you're trying to store large integers */
+  P2033 = HttpStatus.INTERNAL_SERVER_ERROR,
+  /** Transaction failed due to a write conflict or a deadlock. Please retry your transaction */
+  P2034 = HttpStatus.INTERNAL_SERVER_ERROR,
+}
+
+@Catch(Prisma.PrismaClientKnownRequestError)
+export class PrismaClientExceptionFilter extends BaseExceptionFilter {
+  catch(exception: Prisma.PrismaClientKnownRequestError, host: ArgumentsHost) {
+    const statusCode = EPrismaErrorHttpStatusCode[exception.code];
+
+    if (!statusCode || statusCode === HttpStatus.INTERNAL_SERVER_ERROR) {
+      return super.catch(exception, host);
+    }
+
+    const message =
+      `[${exception.code}]: ` +
+      this._generateValidExceptionMessage(exception.message);
```

Comment:
creo que en este caso el message puede ser solo la causa del error (exception.meta.cause), ya que viene dentro de la variable exception y asi se evita hacer todo el parseo del message. Y para logueo se puede colocar la cause junto con el code  `${exception.code} - ${exception.meta.cause}`. Con esto la funcion _generateValidExceptionMessage no es necesaria.

exception:
`{
  code: "P2025",
  clientVersion: "4.5.0",
  meta: {
    cause: "Record to update not found.",
  },
  "\nInvalid `this.prismaService.category.update()` invocation in\nC:\\Users\\franc\\Desktop\\learn bitch\\michis-api\\src\\modules\\category\\category.service.ts:26:40\n\n  23 }\n  24 \n  25 public async updateCategory(id: string, name: string): Promise<Category> {\n→ 26   return this.prismaService.category.update(\nAn operation failed because it depends on one or more records that were required but not found. Record to update not found."
}`

Reply:
La información de la propiedad meta es muy diferente entre los errores de prisma, por ejemplo para:
- not valid constraint: 
meta: {
              target: 'Category_name_key',
}
- Delete not found
 meta: {
              cause: 'Record to delete does not exist.',
}
- Params constraint validation
meta: {
              message:
                'Malformed ObjectID: provided hex string representation must be exactly 12 bytes, instead got: "6358a8a8dd6724c536825ca921", length 26.',
}
Por esa razon considere usar el mensaje y formatearlo porque igual agrupa esas propiedades al mensaje final.